lib/new_relic/agent/new_relic_service.rb

```diff
@@ -19,7 +18,10 @@ class NewRelicService
 
       # These include Errno connection errors, and all indicate that the
       # underlying TCP connection may be in a bad state.
-      CONNECTION_ERRORS = [Timeout::Error, EOFError, SystemCallError, SocketError].freeze
+      CONNECTION_ERRORS = [Net::OpenTimeout, Net::ReadTimeout, EOFError, SystemCallError, SocketError]
+      # Net::WriteTimeout is Ruby 2.6+
```

Comment:
Should we add a todo here to keep a note to update this when 2.5 support is dropped someday? I'm thinking that when we search for things related to 2.5 we can remove we potentially won't notice this one, but if it's a todo we'd probably find it. Same with the other one for WriteTimeout below?

Reply:
Yeah, good call here. Will do.

Reply:
TODOs added with [607f6d5](https://github.com/newrelic/newrelic-ruby-agent/pull/2147/commits/607f6d5c9466bffb725d9151205fd411637f9ad9)